pqarrow/arrow.go

```diff
@@ -359,6 +359,7 @@ func (c *ParquetConverter) NumRows() int {
 
 func (c *ParquetConverter) NewRecord() arrow.Record {
 	if c.builder != nil {
+		defer c.builder.Reset()
```

Comment:
Do we want to make this part of NewRecord?

Reply:
Your call. I don't have a preference

Reply:
I think so. This could come back to bite us if we ever reuse the builder

Reply:
We already reuse the builder. This just causes it to rebuild the dictionaries

Reply:
Made reset an explicit call.